### ISSUE TITLE
Bug: items in refslots dont swap

### ISSUE THREAD

**Issue by kyle-wannacott** [peter-kish/gloot#171]
You can override another refslot with the item from another refslot; which isn't the problem its that they don't swap.

Video:
https://github.com/peter-kish/gloot/assets/49783296/17515b3e-be52-4506-b8da-0d581407e636

 I'm not sure what use case you would need a one to many relationship, but possibly there is (maybe in a hotbar)?

I don't really know what exactly what the behavior should be. However, I guess if you had ring slots you should be able to switch the items in the refSlots. :thinking: 

Note: I'm not using refSlot; so I'm not worried about this issue; was just experimenting and found it.

**Comment by peter-kish** [comment]
Yeah, this is intentional, though I'm also unsure if it was the right decision. It would definitely be useful to be able to swap items (not only in slots, but also in an `InventoryGrid`).

**Comment by peter-kish** [comment]
Item swapping inside inventories and item slots is implemented in [v2.4.6](https://github.com/peter-kish/gloot/releases/tag/v2.4.6), but that still does not cover `ItemRefSlot`s. Since their purpose became somewhat unclear after the introduction of `ItemSlot`s, I'm currently leaning towards deprecating them. For that reason I don't think it makes a lot of sense to introduce new `ItemRefSlot` related logic...

I might replace them at some point with some sort of an "item drop area" control where the user can drop items and connect to some signals to define what happens to them. That would still allow for some custom slots without introducing complex classes with unclear purpose.

**Comment by LeeWannacott** [comment]
> Since their purpose became somewhat unclear

Hmmm, Other than Deus Ex I can't really think of any other game that does the ItemRef thing for inventory. (doesn't mean they don't exist though)

**Comment by peter-kish** [comment]
`ItemRefSlot`s have been removed in v3.0.0.
